x/airdrop/keeper/claim_test.go

```diff
@@ -0,0 +1,450 @@
+	for _, allocation := range allocationsSdkInt {
+		allocationsInt64 = append(allocationsInt64, allocation.Int64())
+	}
+	return allocationsInt64
+}
+
+func (s *KeeperTestSuite) TestClaimDaily() {
+	testCases := []struct {
+		name                string
+		timeOffset          time.Duration
+		initialAllocations  []int64
+		expectedAllocations []int64
+		initialForfeited    int64
+		expectedForfeited   int64
+		initialClaimed      int64
+		expectedClaimed     int64
+		expectedNewRewards  int64
+		initialClaimType    types.ClaimType
+		expectedError       string
+	}{
+		{
+			// 10 rewards accrued on each of 3 days
+			// Claimed shortly into the first day, 10 total claimed
+			name:                "claim the first day",
+			timeOffset:          time.Hour, // one hour into first window
+			initialAllocations:  []int64{10, 10, 10},
+			expectedAllocations: []int64{0, 10, 10},
+			initialClaimed:      100,
+			expectedClaimed:     100 + 10,
+			expectedNewRewards:  10,
+			initialClaimType:    types.CLAIM_DAILY,
+		},
+		{
+			// 10 rewards accrued on each of 3 days
+			// Claimed shortly into the second day, 20 total claimed
+			name:                "claim the second day",
+			timeOffset:          time.Hour * 25, // one hour into second window
+			initialAllocations:  []int64{10, 10, 10},
+			expectedAllocations: []int64{0, 0, 10},
+			initialClaimed:      100,
+			expectedClaimed:     100 + 20,
+			expectedNewRewards:  20,
+			initialClaimType:    types.CLAIM_DAILY,
+		},
+		{
+			// 10 rewards accrued on each of 3 days
+			// Claimed shortly into the second day, 20 total claimed
+			name:                "claim all days at once",
+			timeOffset:          time.Hour * 49, // one hour into third window
+			initialAllocations:  []int64{10, 10, 10},
+			expectedAllocations: []int64{0, 0, 0},
+			initialClaimed:      100,
+			expectedClaimed:     100 + 30,
+			expectedNewRewards:  30,
+			initialClaimType:    types.CLAIM_DAILY,
+		},
+		{
+			// Claimer already chose claim early
+			name:               "already chose to claim early",
+			timeOffset:         time.Hour,
+			initialAllocations: []int64{},
+			initialClaimed:     100,
+			initialClaimType:   types.CLAIM_EARLY,
+			expectedError:      "user has already elected claim option",
+		},
+		{
+			// Claimer has no rewards on the current day
+			name:               "no rewards today",
+			timeOffset:         time.Hour * 49, // one hour into third window
+			initialAllocations: []int64{0, 0, 0, 20},
+			initialClaimed:     100,
+			expectedError:      "no unclaimed rewards",
+		},
+		{
+			// Claimed 1 hour before the airdrop started
+			name:               "airdrop not started yet",
+			timeOffset:         (-1 * time.Hour), // before airdrop start
+			initialAllocations: []int64{},
+			initialClaimed:     100,
+			expectedError:      "airdrop distribution has not started",
+		},
+		{
+			// Claimed well after the airdrop ended
```

Comment:
What about the case where they claim after the airdrop claim window, but have nonzero allocations?

Reply:
I think those two tests are essentially the same since the check on the airdrop window doesn't consider the allocations